File: verdant_erp_backend/app/routes/crm/renewals.py
from fastapi import APIRouter, Depends
from sqlalchemy import select
from sqlalchemy.orm import selectinload
from sqlalchemy.ext.asyncio import AsyncSession
from app.core.database import get_db
from app.models.crm.contract import Contract
from app.schemas.crm.renewal_response import (
    RenewalResponse
)

router = APIRouter(
    prefix="/crm/renewals",
    tags=["CRM Renewals"]
)


@router.get(
    "/",
    response_model=list[RenewalResponse]
)
async def list_renewals(
    db: AsyncSession = Depends(get_db)
):

    result = await db.execute(
        select(Contract)
        .options(
            selectinload(Contract.customer)
        )

        .where(
            Contract.status == "ACTIVE"
        )
    )

    contracts = result.scalars().all()

    return [

        RenewalResponse(
            contract_id=str(c.id),
            customer_id=str(c.customer_id),
            customer_name=(
                c.customer.name
                if c.customer
                else "Unknown Customer"
            ),

            renewal_date=c.renewal_date,
            sla_tier=c.sla_tier,
            total_value=c.total_value,
            status=c.status
        )

        for c in contracts
    ]

File: verdant-frontend/src/lib/api/crm/renewals.ts
import api from '$lib/api/client';

export interface Renewal {
    contract_id: string;
    customer_id: string;
    customer_name: string;
    renewal_date: string | null;
    sla_tier: string | null;
    total_value: number | null;
    status: string | null;
}

export async function getRenewals():

    Promise<Renewal[]> {

    const response =
        await api.get('/crm/renewals');

    return response.data;
}